picard/ui/find.py

```diff
@@ -0,0 +1,156 @@
+        dialog = QtWidgets.QDialog(self)
+        dialog.setWindowTitle(_("Select Filters"))
+        dialog.setMinimumWidth(300)
+
+        layout = QtWidgets.QVBoxLayout(dialog)
+
+        # Scroll area for tags
+        scroll = QtWidgets.QScrollArea(dialog)
+        scroll.setWidgetResizable(True)
+        scroll_content = QtWidgets.QWidget(scroll)
+        scroll_layout = QtWidgets.QVBoxLayout(scroll_content)
+
+        label = QtWidgets.QLabel(_("File Filters"), scroll_content)
+        scroll_layout.addWidget(label)
+
+        # Add a horizontal separator
+        line = QtWidgets.QFrame(scroll_content)
+        line.setFrameShape(QtWidgets.QFrame.Shape.HLine)
+        line.setFrameShadow(QtWidgets.QFrame.Shadow.Sunken)
+        scroll_layout.addWidget(line)
+
+        file_filters = {
+            'filename': N_("Filename"),
+            'filepath': N_("Filepath"),
+        }
+
+        checkboxes = {}
+        for file_filter in ["filename", "filepath"]:
+            checkbox = QtWidgets.QCheckBox(_(file_filters[file_filter]), scroll_content)
+            checkbox.setChecked(file_filter in self.selected_filters)
+            scroll_layout.addWidget(checkbox)
+            checkboxes[file_filter] = checkbox
+
+        scroll_layout.addItem(QtWidgets.QSpacerItem(0, 10, QtWidgets.QSizePolicy.Policy.Minimum, QtWidgets.QSizePolicy.Policy.Fixed))
+
+        label = QtWidgets.QLabel(_("Tag Filters"), scroll_content)
+        scroll_layout.addWidget(label)
+
+        # Add a horizontal separator
+        line = QtWidgets.QFrame(scroll_content)
+        line.setFrameShape(QtWidgets.QFrame.Shape.HLine)
+        line.setFrameShadow(QtWidgets.QFrame.Shadow.Sunken)
+        scroll_layout.addWidget(line)
+
+        # Add checkboxes for all tags
+        for tag in sorted(self.valid_tags):
+            checkbox = QtWidgets.QCheckBox(tag, scroll_content)
+            checkbox.setChecked(tag in self.selected_filters)
+            scroll_layout.addWidget(checkbox)
+            checkboxes[tag] = checkbox
+
+        scroll_content.setLayout(scroll_layout)
+        scroll.setWidget(scroll_content)
+        layout.addWidget(scroll)
+
+        # Buttons
+        button_box = QtWidgets.QDialogButtonBox(
+            QtWidgets.QDialogButtonBox.StandardButton.Ok | QtWidgets.QDialogButtonBox.StandardButton.Cancel)
+        button_box.accepted.connect(dialog.accept)
+        button_box.rejected.connect(dialog.reject)
+        layout.addWidget(button_box)
+
+        # Show dialog and process result
+        if dialog.exec() == QtWidgets.QDialog.DialogCode.Accepted:
+            self.selected_filters = []
+            for tag, checkbox in checkboxes.items():
+                if checkbox.isChecked():
+                    self.selected_filters.append(tag)
+
+            # Update button text
+            if not self.selected_filters or self.selected_filters == []:
+                self.selected_filters = []
+                self.filter_button.setText(_("Filters"))
```

Comment:
Should this default to something like "Select Filters" instead?  I'm okay with it as you have it now, but throwing this out in case it makes the operation clearer and easier for users to understand.